wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -656,6 +667,43 @@ public void deleteProject(@ApiParam("The workspace id")
         }
     }
 
+    @GET
+    @Path("/{id}/check")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Get state of the workspace agent by the workspace id and agent id")
+    @ApiResponses({@ApiResponse(code = 200, message = "The response contains requested workspace entity"),
+                   @ApiResponse(code = 404, message = "The workspace with specified id does not exist"),
```

Comment:
ok for 400 and I think don't need to add for other two (409 and 401),we don't have situations where it can be threw.